File: backend/src/main/java/com/skillforge/service/AdminService.java
package com.skillforge.service;

import com.skillforge.dto.admin.AdminDashboardResponseDto;
import com.skillforge.dto.admin.AdminUserDetailsResponseDto;
import com.skillforge.dto.admin.AdminUserFilterRequestDto;
import com.skillforge.dto.admin.AdminUserStatusRequestDto;
import com.skillforge.dto.admin.AdminUserResponseDto;
import org.springframework.data.domain.Page;

import java.util.UUID;

public interface AdminService {
    AdminDashboardResponseDto getDashboard();

    Page<AdminUserResponseDto> getUsers(AdminUserFilterRequestDto request);

    AdminUserDetailsResponseDto getUserDetails(UUID userId);

    AdminUserDetailsResponseDto updateUserStatus(
            UUID userId,
            AdminUserStatusRequestDto request
    );
}

File: frontend/src/services/adminService.js
import api from "./api";

const unwrap = (response) => response?.data?.data ?? response?.data;

const adminService = {
  getDashboard: async () => {
    const response = await api.get("/admin/dashboard");
    return unwrap(response);
  },

  getUsers: async (params = {}) => {
    const response = await api.get("/admin/users", { params });
    return unwrap(response);
  },

  getUserDetails: async (userId) => {
    const response = await api.get(`/admin/users/${userId}`);
    return unwrap(response);
  },

  updateUserStatus: async (userId, accountStatus) => {
    const response = await api.patch(`/admin/users/${userId}/status`, {
      accountStatus,
    });
    return unwrap(response);
  },
};

export default adminService;
